### PR TITLE
Allow multiple newlines between import pkg and pkg declaration (HPC-7684)

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: lib/vsc/install/shared_setup.py [hunk 1]
```diff
@@ -392,25 +403,39 @@ def rel_gitignore(self, paths, base_dir=None):
...
 
-            patterns = [l.replace('*', '.*') for l in all_patterns if l.startswith('*')]
-            reg = re.compile('^('+'|'.join(patterns)+')$')
+            patterns = [l.replace("*", ".*") for l in all_patterns if l.startswith("*")]
+            reg = re.compile("^(" + "|".join(patterns) + ")$")
 
             # check if we at least filter out .pyc files, since we're in a python project
-            if not all([reg.search(text) for text in ['bla%s' % pattern for pattern in GITIGNORE_PATTERNS]]):
-                raise Exception("%s/.gitignore does not contain these patterns: %s " % (base_dir, GITIGNORE_PATTERNS))
+            if not all(
```
Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: ugly!!

File: lib/vsc/install/shared_setup.py [hunk 2]
```diff
@@ -429,27 +454,39 @@ def files_in_packages(self, excluded_pkgs=None):
...
-                        raise Exception(('vsc namespace packages do not allow non-shared namespace in dir %s.'
-                                         'Fix with pkg_resources.declare_namespace') % root)
-
-                res['packages'][package] = self.rel_gitignore([os.path.join(root, f) for f in files])
+                if "__init__.py" in files and (
+                    package == "vsc" or package.startswith("vsc.")
+                ):
+                    init = open(os.path.join(root, "__init__.py")).read()
+                    if not re.search(
+                        r"^import\s+pkg_resources\n+pkg_resources.declare_namespace\(__name__\)$",
```
Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: i would do `\n{1,3}`, nothing else

File: lib/vsc/install/shared_setup.py (hunk 1 above)
Comment (itkovian) [reply lib/vsc/install/shared_setup.py]: automated!

File: lib/vsc/install/shared_setup.py (hunk 2 above)
Comment (itkovian) [reply lib/vsc/install/shared_setup.py]: all right

**Pull request closed without merging**